test/primitive/measure.jl

```diff
@@ -114,3 +87,30 @@ end
     m = Measure(5)
     @test chmeasureoperator(m, X) == Measure(5, operator = X)
 end
+
+@testset "measure an operator correctly" begin
+    op = kron(Z,Z)
```

Comment:
which case doesn't work previously?

Reply:
This is the case the previous version that does not work...
If you measure `ZZ` operator in the previous version, it will project you to the computational basis.
However, the lastest program will project it to the subspace spanned by `ZZ=1` or `ZZ=-1`.

Reply:
ah make sense!

Reply:
I can show you an example of syndrome detection of Shor's code if you are interested. Where previous measure fails.